Tests/ZIPFoundationTests/ZIPFoundationWritingTests.swift

```diff
@@ -329,4 +329,100 @@ extension ZIPFoundationTests {
         let nonUpdatableArchive = Archive(url: nonUpdatableArchiveURL, accessMode: .update)
         XCTAssertNil(nonUpdatableArchive)
     }
+
+    func testUniqueTemporaryDirectoryURL() {
+        let archive = self.archive(for: #function, mode: .create)
+        var tempURLs = Set<URL>()
+        defer {
+            for url in tempURLs {
+                try? FileManager.default.removeItem(at: url)
+            }
+        }
+        // We choose 2000 temp directories to test workaround for http://openradar.appspot.com/50553219
+        for _ in 1...2000 {
+            autoreleasepool {
+                let tempDir = archive.uniqueTemporaryDirectoryURL()
+                XCTAssertFalse(tempURLs.contains(tempDir), "Temp directory URL should be unique. \(tempDir)")
+                tempURLs.insert(tempDir)
+            }
+        }
+    }
+
+    func testReplaceCurrentArchiveWithArchiveCrossLink() {
+		#if os(macOS)
+        let createVolumeExpectation = expectation(description: "Creation of temporary additional volume")
+        let unmountVolumeExpectation = expectation(description: "Unmount temporary additional volume")
+
+        let tempDir = URL(fileURLWithPath: NSTemporaryDirectory()).appendingPathComponent(UUID().uuidString)
+        do {
+            try FileManager.default.createDirectory(at: tempDir, withIntermediateDirectories: true, attributes: nil)
+        } catch {
+            XCTFail("\(String(describing: error))")
+            return
+        }
+        defer {
+            try? FileManager.default.removeItem(at: tempDir)
+        }
+
+        let volName = "Test_\(UUID().uuidString)"
+        let dmgURL = tempDir.appendingPathComponent(volName).appendingPathExtension("dmg")
+        let script = """
+        #!/bin/bash
+        hdiutil create -size 5m -fs HFS+ -type SPARSEBUNDLE -ov -attach -volname "\(volName)" "\(dmgURL.path)"
+
```

Comment:
Great idea to create a multiple volume environment!

Reply:
Thanks!